pkg/services/govmomi/vcenter/clone.go

```diff
@@ -457,3 +461,25 @@ func getGpuSpecs(ctx *context.VMContext) ([]types.BaseVirtualDeviceConfigSpec, e
 	}
 	return deviceSpecs, nil
 }
+func getVgpuSpecs(ctx *context.VMContext) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	deviceSpecs := []types.BaseVirtualDeviceConfigSpec{}
+	deviceKey := int32(-200)
+
+	expectedVGpuDevices := ctx.VSphereVM.Spec.VirtualMachineCloneSpec.VGPUDevices
+	if len(expectedVGpuDevices) == 0 {
+		return nil, errors.Errorf("Invalid vgpu device count count: %d", len(expectedVGpuDevices))
+	}
+
+	for _, vGPUDevice := range expectedVGpuDevices {
+		backingInfo := &types.VirtualPCIPassthroughVmiopBackingInfo{
+			Vgpu: vGPUDevice.ProfileName,
+		}
+		dynamicDirectPathDevice := createPCIPassThroughDevice(deviceKey, backingInfo)
```

Comment:
is this the "magic" part, i.e. ... we look at a vGPU and make a passthrough device on the fly ?